### PR TITLE
chore: implement garbage collection bdds

### PR DESCRIPTION
**Pull request by Abhinandan-Purkait**

### CODE REVIEW

**Comment by Abhinandan-Purkait** [comment]
bors try

File: tests/bdd/features/snapshot/garbage_collection/test_garbage_collection.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    with open(path, "w") as file:
+        file.truncate(60 * 1024 * 1024)
+
+    yield
+    # Clear the file
+    if os.path.exists(path):
+        os.remove(path)
+
+
+@pytest.fixture(scope="function")
```
Comment (tiagolobocastro) [inline tests/bdd/features/snapshot/garbage_collection/test_garbage_collection.py]: let's use module scope

File: tests/bdd/features/snapshot/garbage_collection/test_garbage_collection.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+            thin=False,
+        ),
+    )
+    if publish_status == "published":
+        ApiClient.volumes_api().put_volume_target(
+            VOLUME1_UUID,
+            publish_volume_body=PublishVolumeBody(
+                {}, Protocol("nvmf"), node=NODE1, frontend_node="app-node-1"
+            ),
+        )
```
Comment (tiagolobocastro) [inline tests/bdd/features/snapshot/garbage_collection/test_garbage_collection.py:99]: yield and delete volume

File: tests/bdd/features/snapshot/garbage_collection/test_garbage_collection.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+    assert len(response.definition.metadata.transactions) == 1
+    assert response.definition.metadata.transactions["2"] is not None
+    assert len(response.definition.metadata.transactions["2"]) == 1
+    assert str(response.definition.metadata.transactions["2"][0]["status"]) == "Created"
+
+
+@then("the snapshot creation should be successful")
+def the_snapshot_creation_should_be_successful():
+    """the snapshot creation should be successful."""
+    assert pytest.exception is None
```
Comment (tiagolobocastro) [inline tests/bdd/features/snapshot/garbage_collection/test_garbage_collection.py:191]: yield and delete snapshot

**Review by tiagolobocastro — APPROVED** [review]

**Comment by Abhinandan-Purkait** [comment]
bors try

**Comment by Abhinandan-Purkait** [comment]
bors merge

**Comment by Abhinandan-Purkait** [comment]
bors merge

**Pull request merged**